lib/transformer.dart

```diff
@@ -84,14 +85,14 @@ class InitializeTransformer extends Transformer {
     if (primaryId == null) primaryId = transform.primaryInput.id;
     var newEntryPointId = new AssetId(primaryId.package,
         '${path.url.withoutExtension(primaryId.path)}.initialize.dart');
-    return transform.hasInput(newEntryPointId).then((exists) {
+    return transform.hasInput(newEntryPointId).then((exists) async{
```

Comment:
What about just changing this to use async/await to replace the `then` calls, instead of inside of them:

```
    Future<AssetId> _buildBootstrapFile(Transform transform,
       {AssetId primaryId}) {
      if (primaryId == null) primaryId = transform.primaryInput.id;
      var newEntryPointId = new AssetId(primaryId.package,
          '${path.url.withoutExtension(primaryId.path)}.initialize.dart');
      var exists = await transform.hasInput(newEntryPointId);
      if (exists) {
        transform.logger
            .error('New entry point file $newEntryPointId already exists.');
        return null;
      }
  
      var resolver = await _resolvers.get(transform, [primaryId]);
      transform.addOutput(generateBootstrapFile(
          resolver, transform, primaryId, newEntryPointId,
          errorIfNotFound: _errorIfNotFound, plugins: plugins));
      resolver.release();
      return newEntryPointId;
    }
```

Ditto for this same issue in other cl's. It should work around the strong mode bugs and clean things up a bit.